core/sql/regress/executor/TEST002

```diff
@@ -1102,7 +1102,7 @@ SELECT student_name,
 
 SELECT student_name,
           GROUP_CONCAT(DISTINCT test_score
-                    ORDER BY test_score SEPARATOR '-')
+                    ORDER BY test_score desc SEPARATOR '-')
```

Comment:
Are we thinking that previously this ORDER BY was actually not doing its job (it was after all guarded by MODE_SPECIAL_4).  It would also be nice if we had a more complex example, just so that we can understand this feature and see it working. This is just a suggestion, please ignore if you do not agree.